### PR TITLE
Don't use the deprecated proton::url

### PR DESCRIPTION
**Pull request by defuse**
Don't use the deprecated proton::url

### CODE REVIEW

**Comment by daira** [comment]
This needs rebasing before further review.

**Comment by defuse** [comment]
Rebased.

**Comment by defuse** [comment]
This makes the proton test hang, looking into it.

**Comment by defuse** [comment]
Fixed the bug, the problem was I wasn't passing in the connection timeout options. `proton_test.py` passes now, this is ready for review.

File: src/amqp/amqpsender.h [hunk 1]
```diff
@@ -15,8 +15,7 @@
 class AMQPSender : public proton::messaging_handler {
   private:
     std::deque<proton::message> messages_; 
-    proton::url url_;
-    proton::connection conn_;
+    std::string url_;
```
Comment (daira) [inline src/amqp/amqpsender.h]: This now doesn't validate the URL when the `AMQPSender` is constructed. Is that a regression in error reporting behaviour? What happens after this PR when an invalid URL is given?

**Comment by daira** [comment]
Obviated by #4588, which removes AMQP/Proton.

**Pull request closed without merging**